addon/components/uni-form.js

```diff
@@ -29,7 +29,10 @@ export default Component.extend({
 
       this.set('isLoading', true);
 
-      this.get('onSubmit')().then(() => this.setIsLoading(false), () => this.setIsLoading(false));
+      let promise = this.get('onSubmit')();
+      if (promise && typeof promise.then === "function") {
```

Comment:
And cause automatic type casts has more costs:

https://jsperf.com/if-variable-of-if-typeof/1